autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
+
+        Returns
+        -------
+        str
+            String representation of query sequence ID, followed by total number of hits and finally the highest
+            bit score of all the hits
+        """
         return f"{self.qseqid}; {len(self.sseqids)} sseqids; top hit by bitscore: {self.get_top_hit()}"
 
     def __eq__(self, other_hit):
+        """
+        Operator overloading to compare two objects of the class
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            Other sseqid hit dictionary to compare with
+
+        Returns
+        -------
+        Boolean
+            True if both dcitionaries are equal else False
+        """
         if self.qseqid == other_hit.qseqid:
             return True
         else:
             return False
 
     def __add__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
+        """
+        Operator overloading to update (add) the sseqids dictionary with the other sseqid hit dictionary
+        for a specific query sequence
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            {
+            sseqid: {
+                "pident": float(pident),
+                "length": int(length),
+                "mismatch": int(mismatch),
+                "gapopen": int(gapopen),
+                "qstart": int(qstart),
+                "qend": int(qend),
+                "sstart": int(sstart),
+                "send": int(send),
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
+            
+        """
+        assert self == other_hit, f"qseqids do not match! {self} & {other_hit}"
         self.sseqids.update(other_hit.sseqids)
         return self
 
     def __sub__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
-        self.sseqids.pop(other_hit.sseqid)
+        """
+        Operator overloading to remove (subtract) the other sseqid hits dictionary from the sseqids dictionary
+        for a specific query sequence
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            {
+            sseqid: {
+                "pident": float(pident),
+                "length": int(length),
+                "mismatch": int(mismatch),
+                "gapopen": int(gapopen),
+                "qstart": int(qstart),
+                "qend": int(qend),
+                "sstart": int(sstart),
+                "send": int(send),
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
+        """
+        assert self == other_hit, f"qseqids do not match! {self} & {other_hit}"
+        self.sseqids.pop(list(other_hit.sseqids.keys())[0])
```

Comment:
This is only removing the first `sseqid` from `self.sseqids`. This may also fail if the `other_hit.sseqids` are not in `self.sseqids`. You should be able to rework this so it will not raise a KeyError when trying to remove `other_hits.sseqids` that are not in the `self.sseqids`. This also needs to be fixed to follow the intended behavior

Reply:
From my understanding of the script and trying with `pdb` I found that `hit.sseqids` will always have a single `sseqid`, and thus we only need to remove the first key. Please let me know in case I am missing something.

Reply:
This may be true for when parsing the blast results, but this may not be true in other circumstances. Only removing one when instead the behavior intends to remove all, could result in a difficiult to find "_gotcha_" kind of 🐛